packages/core/src/rooms.ts

```diff
@@ -0,0 +1,92 @@
+import {Base} from '.'
+
+import {
+  RoomsRequestParameters,
+  RoomsRequestOptions,
+  RoomsResults
+} from './types'
+
+export type RoomsFn = (
+  parameters: RoomsRequestParameters
+) => Promise<RoomsResults>
+
+/**
+ * Creates rooms request string
+ *
+ * @param parameters - rooms request parameters
+ * @param options - rooms request options
+ *
+ * @returns request string
+ */
+function createRoomsRequestString(
+  parameters: RoomsRequestParameters,
+  options: RoomsRequestOptions
+) {
+  const {hotelId, checkIn, checkOut, rooms} = parameters
+
+  const urlParameters = {
+    hotelId,
+    checkIn,
+    checkOut,
+    rooms,
+    deviceType: options.deviceCategory,
+    locale: options.language,
+    currency: options.currency,
+    aid: options.anonymousId
+  }
+
+  return new URLSearchParams(urlParameters).toString()
+}
+
+/**
+ * Generates signature for signing SAPI requests
+ *
+ * @param endpoint - endpoind for which to generate the signature (e.g. "offers-rooms")
+ * @param requestString - request string to generate signature (e.g. url search string)
+ * @param salt - salt
```

Comment:
```suggestion
 * Generates signature for signing SAPI requests.
 *
 * @param endpoint - Endpoint for which to generate the signature (e.g. "offers-rooms")
 * @param requestString - Request string to generate signature (e.g. url search string)
 * @param salt
```

I propose we make the end point a enum to avoid mistakes, and maybe you can add an `@example` too?

Reply:
I'm going to move endpoints to the configuration object